python/kfserving/test/test_storage.py

```diff
@@ -94,61 +96,38 @@ def test_https_uri_path(_):
-
-
-@mock.patch('requests.get', return_value=MockHttpResponse(status_code=200, content_type='application/zip',
-                                                          raw=FILE_ZIP_RAW))
-def test_http_uri_path_with_zip(_):
-    with tempfile.TemporaryDirectory() as out_dir:
-        https_uri = 'https://foo.bar/model.zip'
-        assert kfserving.Storage.download(https_uri, out_dir=out_dir) == out_dir
-        assert os.path.exists(os.path.join(out_dir, 'model.pth'))
-
-
-@mock.patch('requests.get', return_value=MockHttpResponse(status_code=200, content_type='application/zip',
-                                                          raw=FILE_ZIP_RAW))
-def test_http_uri_path_with_zip_query_params(_):
-    with tempfile.TemporaryDirectory() as out_dir:
-        https_with_query_uri = 'https://foo.bar/model.zip?foo=bar'
-        assert kfserving.Storage.download(https_with_query_uri, out_dir=out_dir) == out_dir
-        assert os.path.exists(os.path.join(out_dir, 'model.pth'))
-
-
-@mock.patch('requests.get', return_value=MockHttpResponse(status_code=404))
-def test_nonexistent_uri(_):
-    non_existent_uri = 'https://theabyss.net/model.joblib'
-    with pytest.raises(RuntimeError):
-        kfserving.Storage.download(non_existent_uri)
-
-
-@mock.patch('requests.get', return_value=MockHttpResponse(status_code=200))
-def test_uri_no_filename(_):
-    bad_uri = 'https://foo.bar/test/'
-    with pytest.raises(ValueError):
-        kfserving.Storage.download(bad_uri)
-
-
-@mock.patch('requests.get', return_value=MockHttpResponse(status_code=200, content_type='text/html'))
-def test_html_content_type(_):
-    bad_uri = 'https://some.site.com/test.model'
-    with pytest.raises(RuntimeError):
-        kfserving.Storage.download(bad_uri)
+http_uri_path_testparams = [
+    (HTTPS_URI_TARGZ, MockHttpResponse(200, FILE_TAR_GZ_RAW, 'application/x-tar'), None),
+    (HTTPS_URI_TARGZ, MockHttpResponse(200, FILE_TAR_GZ_RAW, 'application/x-gtar'), None),
+    (HTTPS_URI_TARGZ, MockHttpResponse(200, FILE_TAR_GZ_RAW, 'application/x-gzip'), None),
+    (HTTPS_URI_TARGZ, MockHttpResponse(200, FILE_TAR_GZ_RAW, 'application/gzip'), None),
+    (HTTPS_URI_TARGZ_WITH_QUERY, MockHttpResponse(200, FILE_TAR_GZ_RAW, 'application/x-tar'), None),
+    (HTTPS_URI_TARGZ_WITH_QUERY, MockHttpResponse(200, FILE_TAR_GZ_RAW, 'application/x-gtar'), None),
+    (HTTPS_URI_TARGZ_WITH_QUERY, MockHttpResponse(200, FILE_TAR_GZ_RAW, 'application/x-gzip'), None),
+    (HTTPS_URI_TARGZ_WITH_QUERY, MockHttpResponse(200, FILE_TAR_GZ_RAW, 'application/gzip'), None),
+    ('https://foo.bar/model.zip', MockHttpResponse(200, FILE_ZIP_RAW, 'application/zip'), None),
+    ('https://foo.bar/model.zip?foo=bar', MockHttpResponse(200, FILE_ZIP_RAW, 'application/zip'), None),
+    ('https://theabyss.net/model.joblib', MockHttpResponse(404), RuntimeError),
+    ('https://some.site.com/test.model', MockHttpResponse(status_code=200, content_type='text/html'), RuntimeError),
```

Comment:
If the response is 200, the exception should be None right?

Reply:
These are not new tests. These are equivalent to the [original tests](https://github.com/kubeflow/kfserving/blob/a61d146b5c46706b9e9be85bf6b169262a122c95/python/kfserving/test/test_storage.py#L140-L151). The original tests expect the code to throw exceptions on bad `content_type`/bad URL even with a 200.